src/Integration/MefServices/VsSessionHost.cs

```diff
@@ -298,7 +299,8 @@ private void RegisterLocalServices()
         {
             this.localServices.Add(typeof(ICredentialStoreService), new Lazy<ILocalService>(() => credentialStoreService));
 
-            this.localServices.Add(typeof(IConfigurationProviderService), new Lazy<ILocalService>(() => new ConfigurationProvider(this, credentialStoreService, Logger)));
+            this.localServices.Add(typeof(IObsoleteConfigurationProviderService), new Lazy<ILocalService>(() => new ObsoleteConfigurationProvider(this, credentialStoreService, Logger)));
+            this.localServices.Add(typeof(IConfigurationProviderService), new Lazy<ILocalService>(() => new UnintrusiveConfigurationProvider()));
```

Comment:
The config provider is obtained via a local service provide i.e. the service needs to be registered and created (we'll get rid of this mechanism at some point in the future).

Both the old and new service implementations are registered.